csv_processor.bal

```diff
@@ -262,3 +261,58 @@ function createQuotationFromCSVData(string[][] inputCSVData) returns Quotation[]
     }
     return qutoations;
 }
+
+isolated function createSupplierFromCSVData(string[][] inputCsv) returns Supplier[]|error {
+    Supplier[] suppliers = [];
+    string[] errors = [];
+    foreach [int, string[]] [idx, line] in inputCsv.enumerate() {
+        int csvLineNo = idx + 1;
+        if line.length() != 4 {
+            return error(string `Invalid CSV Length in line:${csvLineNo}`);
+        }
+        var [name, shortName, email, phoneNumber] = readSupplierCsvLine(line);
+        boolean hasErrors = false;
+        if !isValidEmail(email) {
+            errors.push(string `Line:${csvLineNo}| provided email [${email}] is invalid`);
+            hasErrors = true;
+        }
+        phoneNumber = regex:replaceAll(phoneNumber, "\\s+", "");
+        if !isValidPhoneNumber(phoneNumber) {
+            errors.push(string `Line:${csvLineNo}| provided phone number [${phoneNumber}] is invalid`);
+            hasErrors = true;
+        }
+        if !hasErrors {
+            Supplier supplier = {
+                name: name,
+                shortName: shortName,
+                email: email,
+                phoneNumber: phoneNumber
+            };
+            suppliers.push(supplier);
+        }
+    }
+    if errors.length() > 0 {
+        string errorMsg = string:'join(" ", ...errors);
+        return error (errorMsg);
+    }
+    return suppliers;
+}
+
+isolated function readSupplierCsvLine(string[] line) returns [string, string, string, string] => [
+    line[0],
+    line[1],
+    line[2],
+    line[3]
+];
```

Comment:
isn't this same as casting, ie `=> <string[4]>line`, type can also be written as `string[4]`

Reply:
Fixed with [bdd2bd7d115899b95bb7a337423d9f29e4bbd207](https://github.com/LSFLK/MedicinesforLK-AdminAPI/pull/83/commits/bdd2bd7d115899b95bb7a337423d9f29e4bbd207)